core/modules/image/image.tokens.inc

```diff
@@ -0,0 +1,324 @@
+                  // information will be statically cached to avoid needing to
+                  // reload the details for another token.
+                  if (empty($images_style_data[$styled_uri])) {
+                    $images_style_data[$styled_uri] = image_get_info($styled_uri);
+                  }
+
+                  switch ($attribute) {
+                    case 'width':
+                      $output[] = $images_style_data[$styled_uri]['width'];
+                      break;
+
+                    case 'height':
+                      $output[] = $images_style_data[$styled_uri]['height'];
+                      break;
+
+                    case 'mimetype':
+                      $output[] = $images_style_data[$styled_uri]['mime_type'];
+                      break;
+
+                    case 'extension':
+                      $output[] = $images_style_data[$styled_uri]['extension'];
+                      break;
+
+                    case 'filesize':
+                      $output[] = $images_style_data[$styled_uri]['file_size'];
+                      break;
+                  }
+                  break;
+
+                case 'path':
+                  if (isset($explode[2]) && $explode[2] === 'relative') {
+                    $url  = parse_url(image_style_url($type, $field['uri']));
+                    $output[] = $url['path'];
+                  }
+                  else {
+                    $output[] = image_style_url($type, $field['uri']);
+                  }
+                  break;
+
+                default:
+                  if (isset($field[$attribute])) {
+                    $output[] = $field[$attribute];
+                  }
+              }
+            }
+            else {
+              $output[] = image_style_url($type, $field['uri']);
+            }
+          }
+
+          // Compile each of the elements into a comma separated list, per
+          // standard Token practice.
+          $replacements[$original] = implode(', ', $output);
+        }
+      }
+    }
+  }
+
+  return $replacements;
+}
+
+/**
+ * Helper function for getting all available image attributes.
+ *
+ * @return array
+ *   All of the available image attributes.
+ */
+function _image_style_token_image_attributes() {
+  return array(
+    'style_name' => NULL,
+    'uri' => '',
+    'path' => NULL,
+    'path:relative' => NULL,
+    'width' => NULL,
+    'height' => NULL,
+    'extension' => NULL,
+    'filesize' => NULL,
+    'alt' => '',
+    'title' => NULL,
+    'mimetype' => NULL,
+    'first' => '',
+    'second' => '',
+    'third' => '',
+    'last' => '',
+  );
+}
+
+/**
+ * Helper function to generate image style if does not exist.
+ *
+ * @param string $styled_uri
+ *   URI for image style.
+ * @param string $type
+ *   Machine name for image style.
+ * @param string $field_uri
+ *   URI of field to be styled.
+ */
+function _image_generate_image($styled_uri, $type, $field_uri) {
```

Comment:
Because this is in the `image.tokens.inc` file, can you rename this to namespace it specifically for token usage? i.e. `_image_token_generate_image()`

Reply:
renamed.